test/public/envs/overview.test.js

```diff
@@ -177,4 +178,33 @@ module.exports = () => {
         await waitForNavigation(page, () => pressElement(page, '#rowTDI59So3d-runs a'));
         expectUrlParams(page, { page: 'run-detail', runNumber: 103 });
     });
+
+    it('should successfully display dropdown links', async () => {
+        await goToPage(page, 'env-overview');
+
+        // Running env
+        let envId = 'CmCvjNbg';
+        await pressElement(page, `tr[id='row${envId}'] .popover-trigger`);
+        let popover = await page.waitForSelector(`.popover:has(button[id='copy-${envId}'])`);
+        await expectLink(popover, 'a:nth-of-type(1)', {
+            href: 'http://localhost:8081/?q={%22partition%22:{%22match%22:%22CmCvjNbg%22},%22severity%22:{%22in%22:%22W%20E%20F%22}}',
+            innerText: 'Infologger FLP',
+        });
+
+        await expectLink(popover, 'a:nth-of-type(2)', {
+            href: 'http://localhost:8080/?page=environment&id=CmCvjNbg',
+            innerText: 'ECS',
+        });
+
+        // Not running env
+        envId = 'EIDO13i3D';
+        await pressElement(page, `tr[id='row${envId}'] .popover-trigger`);
+        popover = await page.waitForNavigation(`.popover:has(button[id='copy-${envId}'])`);
+        await expectLink(popover, 'a:nth-of-type(1)', {
+            href: 'http://localhost:8081/?q={%22partition%22:{%22match%22:%22EIDO13i3D%22},%22severity%22:{%22in%22:%22W%20E%20F%22}}',
+            innerText: 'Infologger FLP',
+        });
+
+        expect(await popover.$('a:nth-of-type(2)')).to.be.null;
```

Comment:
I think it's better to use `waitForSelector('', {hidden: true});` it seems a bit more robust to me, open to debate.

Reply:
applied